lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -272,6 +272,47 @@ bool isTppZero(linalg::GenericOp linalgOp, SmallVectorImpl<Value> *operands) {
   return true;
 }
 
+SmallVector<Operation *> splitFusedOp(tpp::FusedBrgemmOp fusedBrgemmOp,
+                                      PatternRewriter &rewriter) {
+  assert(fusedBrgemmOp.hasBufferSemantics() &&
+         "tpp.fused_brgemm expects a memref type");
+
+  SmallVector<Operation *> splitOps;
+  Location loc = fusedBrgemmOp.getLoc();
+
+  // Split the fused op into individual operations.
+  auto ins = fusedBrgemmOp.getInputs();
+  auto out = fusedBrgemmOp.getOutput();
+  Operation *brgemm = rewriter.create<tpp::BrgemmOp>(
+      loc, ValueRange{ins[0], ins[1], ins[2]}, out);
+  splitOps.push_back(brgemm);
+
+  switch (fusedBrgemmOp.getBinaryKind()) {
+  case tpp::FusedBinaryOpKind::ADD: {
+    Operation *add =
+        rewriter.create<tpp::AddOp>(loc, ValueRange{ins[3], out}, out);
+    splitOps.push_back(add);
+    break;
+  }
+  case tpp::FusedBinaryOpKind::NONE: {
+    break;
+  }
+  }
+
+  switch (fusedBrgemmOp.getUnaryKind()) {
+  case tpp::FusedUnaryOpKind::RELU: {
+    Operation *relu = rewriter.create<tpp::ReluOp>(loc, out, out);
+    splitOps.push_back(relu);
+    break;
+  }
+  case tpp::FusedUnaryOpKind::NONE: {
+    break;
+  }
+  }
+
+  return splitOps;
```

Comment:
There are no early exits, no branches, nothing that would require careful control flow analysis to only remove the old op in special cases. You can just insert the removal here.